### PR TITLE
ref(integrations): Allow migrating repos via dropdown

### PR DESCRIPTION
**Pull request by MeredithAnya**
* Adds the frontend piece for https://github.com/getsentry/sentry/pull/9978
* Makes sure the migratable repo shows up in the dropdown

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx [hunk 1]
```diff
@@ -184,8 +192,10 @@ export default class IntegrationRepos extends AsyncComponent {
         </DropdownButton>
       );
     }
-    const repositories = new Set(this.state.itemList.map(i => i.name));
-    const repositoryOptions = (this.state.integrationRepos.repos || []).filter(
+    const repositories = new Set(
+      this.state.itemList.filter(item => item.integrationId !== 'None').map(i => i.name)
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx]: Why is it the string `None`?

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx [hunk 2]
```diff
@@ -99,16 +99,24 @@ export default class IntegrationRepos extends AsyncComponent {
     let {integration} = this.props;
     let orgId = this.context.organization.slug;
     let {itemList} = this.state;
+    let migratableRepo = itemList.filter(item => selection.value === item.name)[0];
+    let path = migratableRepo
+      ? `/organizations/${orgId}/repos/${migratableRepo.id}/`
+      : `/organizations/${orgId}/repos/`;
+    let data = migratableRepo
+      ? {integrationId: integration.id}
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx]: This gets handled here, once we have the integration https://github.com/getsentry/sentry/pull/9978/files#diff-6e3a10ff3614a2910f4149aa6e5abe1aR73

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx (hunk 1 above)
Comment (MeredithAnya) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx]: This seems to be how it comes back when the integrationId isn't set

Comment (benvinegar) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx]: Can we figure this out? The backend should be returning null. cc @mnoble

File: src/sentry/api/serializers/models/repository.py [hunk 3]
```diff
@@ -9,6 +9,9 @@
 @register(Repository)
 class RepositorySerializer(Serializer):
     def serialize(self, obj, attrs, user):
+        integration_id = None
+        if obj.integraiton_id:
```
Comment (mnoble) [inline src/sentry/api/serializers/models/repository.py]: `integraiton_id` --> `integration_id` (typo)

**Review by mnoble — CHANGES_REQUESTED** [review]
Small typo fix, otherwise looks good!

File: src/sentry/api/serializers/models/repository.py (hunk 3 above)
Comment (MeredithAnya) [reply src/sentry/api/serializers/models/repository.py]: damn

**Review by mnoble — APPROVED** [review]
Good stuff! 👏

**Pull request merged**